### DISCUSSION TITLE
Add Core plugins

### DISCUSSION THREAD

**Discussion by Jopp-gh** [Ulauncher/Ulauncher#1409] [Ideas]
Hi to all,
I was wondering why Ulauncher doesn't ship with some **core plugins**, in other words, some plugins which work all times no matter what version of ULauncher we use.

I'm talking about 
https://github.com/kamilfurman/ulauncher-text-expander

snippets are super useful when you write:
- emails
- code
- forum posts
- etc

Unfortunately, **ulauncher-text-expander** loads with errors and isn't usable anymore, probably because it needs some smaller update to work again. I've no idea about python, I only know bash scripting. 

Not to complicate this post, but a **calculator** should be included in core plugins as well.

**Comment by friday** [comment]
Calculator mode has been supported in Ulauncher even before Ulauncher supported extensions. Just type math into the Ulauncher.

If you need a specific extension it's on you to make it, or fork and fix a similar extension, or pay someone to do it for you. We don't want to support every possible use case users want. That's why we have extensions to begin with. I don't use text expanders, so I would not be a good maintainer for that. And why not use a standalone text expander? Have you tried https://espanso.org/, the most popular one?
